app/controllers/bars_controller.rb

```diff
@@ -18,6 +23,43 @@ def create
     end
   end
 
+  def show
+    @bar = Bar.find(params[:id])
+    @current_user = current_user
+  end
+
+  def edit
+    if user_signed_in?
+      @bar = Bar.find(params[:id])
+    else
+      flash[:notice] = "You must be signed in to do that"
+      redirect_to bar_path(params[:id])
+    end
+  end
+
+  def update
+    @bar = Bar.find(params[:id])
+    if @bar.update(bar_params)
+      flash[:notice] = "Bar updated!"
+      redirect_to bar_path(@bar.id)
+    else
+      flash[:notice] = @bar.errors.full_messages.join(" ")
+      render 'edit'
+    end
+  end
+
+  def destroy
+    if current_user.admin?
+      @bar = Bar.find(params[:id])
+      @bar.destroy
```

Comment:
The reviews associated with `@bar` should probably be deleted too since the parent has been destroyed. Look into `dependent: :destroy`:

http://guides.rubyonrails.org/association_basics.html

It automaticaly deletes all the children associations when a parent is destroyed.